### PR TITLE
Update build information to retrieve vcsroot.branch

### PR DESCRIPTION
**Pull request by paulegradie**
Fixes: https://github.com/OctopusDeploy/Issues/issues/6766

### CODE REVIEW

File: octopus-agent/src/main/java/octopus/teamcity/agent/OctopusBuildInformationBuildProcess.java [hunk 1]
```diff
@@ -63,7 +63,7 @@ protected OctopusCommandBuilder createCommand() {
                     sharedConfigParameters.get("octopus_vcstype"),
                     sharedConfigParameters.get("octopus_vcsroot"),
                     sharedConfigParameters.get("build.vcs.number"),
-                    sharedConfigParameters.get("octopus_branch"),
+                    sharedConfigParameters.get("vcsroot.branch"),
```
Comment (andyinaus) [inline octopus-agent/src/main/java/octopus/teamcity/agent/OctopusBuildInformationBuildProcess.java:66]: This feels like we didn't assign `vcsroot.branch` to `octopus_branch` parameter properly somewhere

**Comment by paulegradie** [comment]
Good point @andyinaus -- I see we are setting these values... but I don't quite understand why we need to do that. Will reach out to someone planning to be on the integrations team and sus things out.

**Comment by twerthi** [comment]
Any update on this?

**Comment by paulegradie** [comment]
@slewis74

**Comment by slewis74** [comment]
I did some testing on my instance and this change would not yield the same answer we currently get. E.g.
image
image
I believe this has to do with the API we use in `OctopusBuildInformationBuildStartProcessor`, as @andyinaus mentioned, but also how the branch filters are configured in the VCS root.
It would be a breaking change for existing customers, they could start getting very different branch names feeding through to Octopus.
As an example, a branch filter of `+:refs/heads/*` yields the branch name shown above. It also looks like this in TeamCity
image

**Comment by paulegradie** [comment]
Makes sense. Thanks for the followup investigation. Going to close out this PR.

**Pull request closed without merging**